lib/prometheus_exporter/instrumentation/sidekiq.rb

```diff
@@ -69,19 +78,30 @@ def get_job_wrapper_name(msg)
     end
 
     def get_delayed_name(msg, class_name)
-      # fallback to class_name since we're relying on the internal implementation
-      # of the delayed extensions
-      # https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/extensions/class_methods.rb
       begin
+        # fallback to class_name since we're relying on the internal implementation
+        # of the delayed extensions
+        # https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/extensions/class_methods.rb
         (target, method_name, _args) = YAML.load(msg['args'].first) # rubocop:disable Security/YAMLLoad
```

Comment:
We could use [`unsafe_load`](https://docs.ruby-lang.org/en/3.1/Psych.html#method-c-unsafe_load) here in case this is ruby 3.1 / `unsafe_load` exists, would make the code easier. `unsafe_load` is used in any ruby version < 3.1 as the default anyway

Reply:
I did originally try to go with that approach but it kept hitting errors with the older ruby versions. YAML.load worked for all versions so I didn't investigate further. Now I look it seems like Psych.unsafe_load was introduced fairly recently as a way to help people who want to upgrade to ruby 3.1 but still need to use the unsafe load instead of the new default.